core/src/main/java/org/elasticsearch/discovery/zen/ZenDiscovery.java

```diff
@@ -35,6 +38,7 @@
 import org.elasticsearch.cluster.routing.allocation.AllocationService;
 import org.elasticsearch.cluster.routing.allocation.RoutingAllocation;
 import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.cluster.service.DuplicateClusterStateUpdateTaskException;
```

Comment:
The exception is not really used in this class? So far it's only used in the tests? If it's only used by the test then I would prefer we just match on the message of the IllegalArgumentException and not introduce this exception.